inst/include/mode/mode.hpp

```diff
@@ -55,6 +55,15 @@ class container {
     }
   }
 
+  void set_state(const std::vector<double>& state) {
+    const bool individual = state.size() == n_state() * n_particles_;
+    const size_t n = individual ? 1 : n_particles_;
+    auto it = state.begin();
+    for (size_t i = 0; i < n_particles_; ++i) {
+      solver_[i].set_state(it + (i / n) * n_state());
+    }
+  }
+
   void set_time(double time) {
```

Comment:
Below, in set_time, we should not be setting `y` in the case where the user has provided state.  So this function probably needs a `set_initial_state` state argument; see the dust code for this, though our case is a bit nastier.

Probably we need to defer the time setting until the last part of the update function if it will control setting things like initial step sizes etc?

Reply:
Ah yeah, of course, sort of forgot about this interaction. Ok, will try and re-jig this and add the `reset_step_size` argument